app/views/conversations/show.html.erb

```diff
@@ -1,15 +1,100 @@
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+
+        <div class="text-white text-opacity-40 text-xs mx-2 mt-10 mb-2">
+          Yesterday
+        </div>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+        <%= render 'conversation_sidebar', title: "Custom opacity in tailwind" %>
+      </div>
+
+      <div id="left-footer" class="flex-none flex text-white mb-4 mt-3">
+        <div class="h-10 flex pr-2 rounded-lg hover:bg-zinc-400 hover:bg-opacity-25 justify-between items-center w-full group cursor-pointer mb-1">
+          <div id="profile" class="ml-2 mr-2 w-7 h-7 bg-white rounded-full"></div>
+          <div class="flex-1 text-white truncate">
+            Keith Schacht
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+
+
+  <div id="right-column" class="flex flex-1 flex-col h-screen bg-white dark:bg-graphite">
+    <!-- Header that does not grow -->
+    <div id="wide-header" class="flex-none justify-between ml-2 mt-2 mb-3 h-10 text-black dark:text-white text-lg hidden md:flex">
+      <%= render 'gpt_header', conversation: @conversation %>
+
+      <div id="right-aligned-content" class="hover:bg-gray-100 cursor-pointer inline-flex mr-5 border border-gray-300 rounded-lg px-2 py-2 mt-1 tooltip tooltip-bottom hover:tooltip-open" data-tip="Share">
+        <%= icon "arrow-up-tray", variant: :outline, size: 18, class: 'text-black' %>
+      </div>
+    </div>
+
+    <div id="narrow-header" class="flex-none h-14 text-lg text-center bg-white dark:bg-zinc-800 md:hidden">
+      <div class="flex-1 mb-2 border-b border-zinc-300 dark:border-zinc-600 text-black dark:text-white dark:bg-graphite flex items-center justify-center">
+        <%= render 'gpt_header', conversation: @conversation %>
+      </div>
+    </div>
+
+    <div id="right-content-container" class="flex flex-col flex-grow overflow-auto">
+      <div id="right-content" class="flex-grow overflow-auto">
+        <div class="w-full md:max-w-none lg:max-w-[700px] xl:max-w-[810px] mx-auto mt-2">
+          <turbo-frame id="messages" src="<%= conversation_messages_path(@conversation) %>">
+            Loading...
+          </turbo-frame>
+        </div>
+      </div>
+
+      <div id="right-footer" class="flex-none flex flex-col relative">
+        <div class="text-center absolute -top-14 left-1/2 -ml-9">
```

Comment:
Even without tailwind a negative margin scares me a little. What's up?

Reply:
This was tricky, but I couldn’t figure out another way. Are negative margins inherently bad? Is there a “gotcha” with it that I may not be considering?

Basically, when the center column of chat messages is not scrolled to the very bottom, a down arrow appears and it floats above the chat input.

So this down arrow isn’t really part of the chat input but it’s position is determined relative to the chat input. It was really hard to figure out how to make this part of the overall scrolling area but be anchored at the bottom. But now that I write this out, I just thought of another way I could try.